public/docs/ts/latest/cookbook/third-party-lib.jade

```diff
@@ -0,0 +1,247 @@
+
+  Normally third party libraries will be published to `npm`. This cookbook simulates publishing by executing all required steps in the `publish.js` script.
+
+  `publish.js` will do the following:
+
+  1) AoT compile the Hero-Profile library by running `node_modules/.bin/ngc -p tsconfig-aot.json`
+
+  2) Inline templates and css using the `inline-resources.js` script
+
+  3) Create an `umd` bundle by running `node_modules/.bin/rollup -c rollup-config.js`
+
++makeTabs(
+  `cb-third-party-lib/hero-profile/publish.js,
+  cb-third-party-lib/hero-profile/inline-resources.js`,
+  null,
+  `publish.js,
+  inline-resources.js`
+)(format='.')    
+
+.l-main-section
+<a id="integrate-with-app"></a>
+:marked
+  ## Integrate with Application
+
+  The library is now ready to be integrated with either AoT compiled applications or JiT compiled applications. The following sections describes how to configure both.
+
+  For the purposes of this demo the JiT and the AoT versions are loaded using the same index.html page. 
+
+  ### AoT
+
+  AoT compiled applications will integrate the library in the compilation of the application as a whole.
+
+  As in the <a href="/docs/ts/latest/cookbook/aot-compiler.html">AoT compilation Cookbook</a> `ngc` is used in combination with `Rollup` to AoT and Tree Shake the application.
+
+  Run the command `ngc -p tsconfig-aot.json && rollup -c rollup-config.js` to execute the combined steps of AoT compilation and Tree Shaking.
+
+  `tsconfig-aot.json` and `rollup-config.js` contain the necessary configuration.
+
++makeTabs(
+  `cb-third-party-lib/ts/rollup-config.js,
+   cb-third-party-lib/ts/tsconfig-aot.json`,
+  null,
+  `rollup-config.js,
+   tsconfig-aot.json`
+)(format='.')
+
+:marked
+  Inside `rollup-config` there is a `nodeResolve` section. This is where the `module` setting from `package.json` comes into play.
+
+  `nodeResolve` will look for either `module` or `jsnext` in `package.json` to determine how to bundle external libraries.
+
++makeExample('cb-third-party-lib/ts/rollup-config.js', 'nodeResolve', 'rollup-config.js')(format=".")
+
+:marked
+  The combined output of `ngc` and `Rollup` is a single `build.js` JavaScript file. 
+  
+  `build.js` contains the entire application including all Angular dependencies and the third party Hero-Profile library.
+
+  To run the application in AoT mode, include `build.js` as a script tag and the `my-aot-app` root level component tag in `index.html`
+
+  ### JiT
+
+  JiT applications load the `umd` bundle using `SystemJS`. This requires a minor tweak to `systemjs.config.js` to register the `umd` bundle with `SystemJS`.
+
+  Simply add `'hero-profile': 'npm:hero-profile/bundles/hero-profile.umd.js'` to the map section of `systemjs.config.js`. 
+
+.l-main-section
+<a id="final-app"></a>
+:marked
+   ## Final Application 
+
+   If you have cloned the `Angular.io` repo, all the steps described in this cookbook can be executed by running the following command:
```

Comment:
I think that makes sense. Some of the complexity comes from having to simulate an npm dependency, and at the same time, make it work with the angular.io workflow. 

I think most of the value is in teaching what types of files are necessary to make the library AoT compatible.